OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1805,23 +1810,24 @@ void Model::setControls(const SimTK::State& s, const SimTK::Vector& controls) co
 const Vector& Model::getControls(const SimTK::State &s) const
 {
     if( (!_system) || (!_modelControlsIndex.isValid()) ){
-        throw Exception("Model::getControls() requires an initialized Model./n" 
+        throw Exception("Model::getControls() requires an initialized Model./n"
             "Prior call to Model::initSystem() is required.");
     }
 
-    // direct the system shared cache 
-    Measure_<Vector>::Result controlsCache = Measure_<Vector>::Result::getAs(_system->updDefaultSubsystem().getMeasure(_modelControlsIndex));
-
-    if(!controlsCache.isValid(s)){
-        // Always reset controls to their default values before computing controls
-        // since default behavior is for controllors to "addInControls" so there should be valid
-        // values to begin with.
-        controlsCache.updValue(s) = _defaultControls;
-        computeControls(s, controlsCache.updValue(s));
-        controlsCache.markAsValid(s);
-    }
-
-    return controlsCache.getValue(s);
+    // Measure_<Vector>::Result controlsCache = Measure_<Vector>::Result::getAs(_system->updDefaultSubsystem().getMeasure(_modelControlsIndex));
+
+    //TODO: Is this necessary?
+    // if(!controlsCache.isValid(s)){
+    //      // Always reset controls to their default values before computing controls
+    //      // since default behavior is for controllors to "addInControls" so there should be valid
+    //      // values to begin with.
+    //      controlsCache.updValue(s) = _defaultControls;
+    //      computeControls(s, controlsCache.updValue(s));
+    //      controlsCache.markAsValid(s);
+    //  }
```

Comment:
I wasn't sure it was necessary to check the validity of the controlsCache during the realization - I was thinking that if they weren't valid we should throw an exception?